substrate/frame/scheduler/src/lib.rs

```diff
@@ -148,6 +159,24 @@ pub struct Scheduled<Name, Call, BlockNumber, PalletsOrigin, AccountId> {
 	_phantom: PhantomData<AccountId>,
 }
 
+impl<Name, Call, BlockNumber, PalletsOrigin, AccountId>
+	Scheduled<Name, Call, BlockNumber, PalletsOrigin, AccountId>
+where
+	Call: Clone,
+	PalletsOrigin: Clone,
+{
+	pub fn as_retry(&self) -> Self {
+		Self {
+			maybe_id: None,
+			priority: self.priority,
+			call: self.call.clone(),
+			maybe_periodic: None,
```

Comment:
now I wonder what if retry will be scheduled as periodic task, would it simplify the solution

Reply:
You would still need a way to tell if the task is a retry or a regular task, because a retry task needs to be dropped after it runs once successfully. I think the current solution is good enough for now after addressing all comments.

Reply:
yeah, I thought about it over night, it does not make sense to make it as periodic